server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Capability.scala

```diff
@@ -0,0 +1,89 @@
+/** *************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.model
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.collection.NonEmpty
+import eu.timepit.refined.string.Uri
+import org.apache.james.jmap.model.CapabilityIdentifier.{CapabilityIdentifier, JMAP_CORE, JMAP_MAIL}
+import org.apache.james.jmap.model.CoreCapabilityProperties.CollationAlgorithm
+import org.apache.james.jmap.model.MailCapability.EmailQuerySortOption
+import org.apache.james.jmap.model.UnsignedInt.UnsignedInt
+
+object CapabilityIdentifier {
+  type CapabilityIdentifier = String Refined Uri
+  val JMAP_CORE: CapabilityIdentifier = "urn:ietf:params:jmap:core"
+  val JMAP_MAIL: CapabilityIdentifier = "urn:ietf:params:jmap:mail"
+}
+
+sealed trait CapabilityProperties
+
+sealed trait Capability {
```

Comment:
IMO that should not be sealed: I can very well imagine write my own Capabilities in extensions